apps/service_providers/models.py

```diff
@@ -164,6 +180,79 @@ def get_speech_service(self) -> speech_service.SpeechService:
         config = {k: v for k, v in self.config.items() if v}
         return self.type_enum.get_speech_service(config)
 
+    def validate_uploaded_file_names(self, uploaded_files: list[File]):
+        """Validate file types. Raises UserServiceProviderConfigError for invalid file types"""
+        if self.type == VoiceProviderType.openai_voice_engine:
+            accepted_file_types = ["mp4", "mp3"]
+            invalid_extentions = set()
+            for file in uploaded_files:
+                file_extention = file.name.split(".")[1]
+                if file_extention not in accepted_file_types:
+                    invalid_extentions.add(f".{file_extention}")
+            if invalid_extentions:
+                string = ", ".join(invalid_extentions)
+                raise UserServiceProviderConfigError(f"File extentions not supported: {string}")
+
+    @transaction.atomic()
+    def add_files(self, files):
+        self.validate_uploaded_file_names(files)
+        if self.type == VoiceProviderType.openai_voice_engine:
+            for file in files:
+                try:
+                    # TODO: Split file extention
+                    SyntheticVoice.objects.create(
+                        name=file.name,
+                        neural=True,
+                        language="",
+                        language_code="",
+                        gender="",
+                        service=SyntheticVoice.OpenAIVoiceEngine,
+                        voice_provider=self,
+                        file=file,
+                    )
+                except IntegrityError:
+                    message = f"Unable to upload '{file.name}' voice. This voice might already exist"
+                    raise UserServiceProviderConfigError(message)
+
+    def remove_file(self, file_id: int):
+        synthetic_voice = self.syntheticvoice_set.get(file_id=file_id)
+        synthetic_voice.file.delete()
+        synthetic_voice.delete()
+
+    def get_files(self):
+        """Return the files found on the synthetic voices that points to this instance"""
+        files = []
+        for voice in self.syntheticvoice_set.all():
```

Comment:
what about adding a filter for file:
```suggestion
        for voice in self.syntheticvoice_set.filter(file__isnull=False).all():
```

Reply:
https://github.com/dimagi/open-chat-studio/pull/376/commits/a1562a726747ab7fcbe5940b61719dabd03618f4